src/som/compiler/Parser.java

```diff
@@ -1444,6 +1450,31 @@ private LiteralNode literalString() throws ParseError {
     return new StringLiteralNode(s, getSource(coord));
   }
 
+  private LiteralNode literalArray(final MethodBuilder builder) throws ProgramDefinitionError {
+    SourceCoordinate coord = getCoordinate();
+    List<ExpressionNode> expressions = new ArrayList<ExpressionNode>();
+
+    expect(NewLiteralArray, DelimiterOpeningTag.class);
+
+    boolean sawPeriod = true;
+
+    while (true) {
+      comments();
+
+      if (sym == EndLiteralArray) {
+        expect(EndLiteralArray, DelimiterClosingTag.class);
+        return createLiteralArray(expressions, getSource(coord));
+      }
+
+      if (!sawPeriod) {
+        expect(Period, null);
+      }
+
+      expressions.add(expression(builder));
+      sawPeriod = accept(Period, StatementSeparatorTag.class);
+    }
```

Comment:
You're right, I missed that case.

Comments and expressions are the only thing that's supported by the grammar.